### PR TITLE
feat(clustering) pre-create the config cache file with 0600 permission

### PR DESCRIPTION
**Pull request by dndx**
on startup

### CODE REVIEW

File: spec/02-integration/09-hybrid_mode/01-sync_spec.lua [hunk 1]
```diff
@@ -175,6 +175,14 @@ for _, strategy in helpers.each_strategy() do
           end
         end, 5)
       end)
+
+      it("local cached config file has correct permission", function()
+        local handle = io.popen("stat servroot2/config.cache.json.gz")
```
Comment (javierguerragiraldez) [inline spec/02-integration/09-hybrid_mode/01-sync_spec.lua:180]: `stat` isn't POSIX, it's part of the GNU coreutiils.  do we run on any BSD system?  (i guess Darwin does install GNU tools now)

**Comment by eskerda** [comment]
Just thinking out loud, will `update_config` thread also always run on the master process?

File: spec/02-integration/09-hybrid_mode/01-sync_spec.lua (hunk 1 above)
Comment (hishamhm) [reply spec/02-integration/09-hybrid_mode/01-sync_spec.lua:180]: > do we run on any BSD system?

Some devs run the test suite on macOS. `stat` flavors are always a portability annoyance. A simple `ls -l` should be enough for the purposes of this test.

**Comment by javierguerragiraldez** [comment]
> will `update_config` thread also always run on the master process?

seems so, since the local `communicate()` is only entered from within the `if ngx.worker.id() == 0 then` block.

File: spec/02-integration/09-hybrid_mode/01-sync_spec.lua (hunk 1 above)
Comment (dndx) [reply spec/02-integration/09-hybrid_mode/01-sync_spec.lua:180]: Actually on macOS 10.15, `stat` is already pre-installed, so it should not pose any challenges for running on macOS.

```
~ datong$ which stat
/usr/bin/stat
```

Comment (dndx) [reply spec/02-integration/09-hybrid_mode/01-sync_spec.lua]: Changed to `ls -l`.

**Review by bungle — APPROVED** [review]

**Pull request merged**